packages/ctt/src/ctc.ts

```diff
@@ -0,0 +1,139 @@
+import { CalmsTypescriptBase, CalmsTypescriptBaseOptions } from './ctb';
+import { ManagedJsonFile } from './managed-json-file';
+
+export interface CalmsTypescriptCdkOptions extends CalmsTypescriptBaseOptions {}
+
+export class CalmsTypescriptCdk extends CalmsTypescriptBase {
+  constructor(options: CalmsTypescriptCdkOptions) {
+    super(options);
+
+    // Add CDK dev dependencies
+    this.addDevDeps('aws-cdk-lib', 'aws-cdk');
+
+    // Override the compile task to use tsc --noEmit
+    const compileTask = this.tasks.tryFind('compile');
+    if (compileTask) {
+      compileTask.reset('tsc -p tsconfig.json --noEmit');
+    }
+
+    // Add pre-compile task to clean build and cdk.out directories
+    const preCompileTask = this.tasks.tryFind('pre-compile');
```

Comment:
Just use this.preCompileTask here and in all the other places - always prefer using the available instance tasks over using tryFind